### PR TITLE
Crash reporting

### PR DESCRIPTION
**Pull request by azlindsay**

### CODE REVIEW

File: Sources/Kumulos.swift [hunk 1]
```diff
@@ -46,7 +47,7 @@ open class Kumulos {
 
     fileprivate(set) var apiKey: String
     fileprivate(set) var secretKey: String
-
+
```
Comment (cgwyllie) [inline Sources/Kumulos.swift]: whitespace

File: KSConfig.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        _apiKey = apiKey
+        _secretKey = secretKey
+        _enableCrash = enableCrash
+    }
+    
+    private var _apiKey: String
+    private var _secretKey: String
+    private var _enableCrash: Bool
+    
+    var ApiKey: String {
```
Comment (cgwyllie) [inline KSConfig.swift]: Fields & methods should be `lowerCamelCase`

Comment (azlindsay) [reply KSConfig.swift]: Done.

File: Sources/Kumulos+Crash.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+//  KumulosSDK
+//
+//  Created by Andy on 26/06/2017.
+//  Copyright © 2017 Kumulos. All rights reserved.
+//
+
+import Foundation
+import KSCrash
+
+public extension Kumulos {
```
Comment (cgwyllie) [inline Sources/Kumulos+Crash.swift]: Does this need to be public / can be internal?

Are we going to expose a helper such as `logException` to allow sending non-fatal reports?

File: KSConfig.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        _enableCrash = false
+    }
+    
+    public func enableCrash() -> KSConfigBuilder {
+        _enableCrash = true
+        return self;
+    }
+    
+    public func build() -> KSConfig {
+        return KSConfig(apiKey: _apiKey, secretKey: _secretKey, enableCrash: _enableCrash)
```
Comment (cgwyllie) [inline KSConfig.swift]: We seem to have lost some assertions on empty credentials, would here be a good place to reintroduce them?

Comment (cgwyllie) [inline KSConfig.swift]: We seem to have lost some assertions on empty credentials, would here be a good place to reintroduce them?

File: Sources/Kumulos+Crash.swift (hunk 3 above)
Comment (cgwyllie) [inline Sources/Kumulos+Crash.swift]: Does this need to be public / can be internal?

Are we going to expose a helper such as logException to allow sending non-fatal reports?

File: KSConfig.swift (hunk 4 above)
Comment (azlindsay) [reply KSConfig.swift]: These were only for the plist initialiser which has been deprecated.

File: Sources/Kumulos+Crash.swift (hunk 3 above)
Comment (azlindsay) [reply Sources/Kumulos+Crash.swift]: The log crashes function is now internal.

File: KSConfig.swift (hunk 4 above)
Comment (cgwyllie) [reply KSConfig.swift]: ok

**Pull request merged**